JavaScript/Lesson27/script.js

```diff
@@ -0,0 +1,180 @@
+console.log('hello world!');
+function task1() {
+
+  let element = document.getElementById('name-task');
+  element.innerText = 'Task 1';
+  let a, b;
+
+  function enter() {
+    a = prompt('Enter first number: ');
+    b = prompt('Enter second number: ');
+
+    if (a === null || b === null) {
+      document.getElementById('canceled-operation').innerHTML = 'Operation canceled!';
+      return;
+    }
+
+    if (a === '' || b === '') {
+      alert('Error! You entered an empty field, please enter again');
+      enter();
+    } else {
+      a = Number(a);
+      b = Number(b);
+      if (isNaN(a) || isNaN(b)) {
+        alert('Error! Entered value is not a number, please enter again');
+        enter();
+      } else {
+        if (a < b) {
+          let confirmation = confirm('Are you sure you want to proceed with the operation?');
+          if (confirmation) {
+            let resultDiff = diff(a, b);
+            document.getElementById('result-diff').innerHTML = `Diff: <span class = "result-diff-value-vid">${resultDiff}</span>`;
+          } else {
+            document.getElementById('canceled-operation').innerText = 'You canceled the operation!';
+          }
+        } else {
+          showResults(a, b);
+        }
+      }
+    }
+  }
+
+  function sum(a, b) {
+    return a + b;
+  }
+
+  function diff(a, b) {
+    return a - b;
+  }
+
+  function add(a, b) {
+    return a * b;
+  }
+
+  function div(a, b) {
+    if (Number(b) === 0) {
+      document.getElementById('error-text').innerText = 'Error! Cannot be divided by zero';
+      return;
+    }
+    return a / b;
+  }
+
+  function showResults(a, b) {
+    let resultSum = sum(a, b);
+    let resultDiff = diff(a, b);
+    let resultAdd = add(a, b);
+    let resultDiv = div(a, b);
+
+    if (typeof resultDiv !== 'undefined') {
+
+      function getColorWithOpacity(value) {
```

Comment:
Why if I only used it once?

Reply:
And what should be done in this case?
Move the function outside the "showResults()" function?
Should it be renamed so that its purpose is visible, although it seems that it already carries an understandable name

Reply:
yes, move outside the "showResults()" function

we have never declared function inside of function, that's a bad practice
functions shouldn't be too complicated